File: classfields/public-api-tests/src/test/java/ru/auto/tests/publicapi/favorites/DeleteSubscriptionTest.java
package ru.auto.tests.publicapi.favorites;

import com.carlosbecker.guice.GuiceModules;
import com.carlosbecker.guice.GuiceTestRunner;
import com.google.inject.Inject;
import io.qameta.allure.junit4.DisplayName;
import org.junit.Ignore;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.RuleChain;
import org.junit.runner.RunWith;
import ru.auto.tests.passport.account.Account;
import ru.auto.tests.publicapi.ApiClient;
import ru.auto.tests.publicapi.adaptor.PublicApiAdaptor;
import ru.auto.tests.publicapi.module.PublicApiSearchesModule;

import static org.apache.http.HttpStatus.SC_FORBIDDEN;
import static ru.auto.tests.commons.restassured.ResponseSpecBuilders.shouldBeCode;
import static ru.auto.tests.commons.restassured.ResponseSpecBuilders.validatedWith;
import static ru.auto.tests.publicapi.model.AutoApiOffer.CategoryEnum.CARS;
import static ru.auto.tests.publicapi.ra.RequestSpecBuilders.defaultSpec;
import static ru.auto.tests.publicapi.ra.ResponseSpecBuilders.shouldBe401NeedAuthentication;

/**
 * Created by vicdev on 22.09.17.
 */

@DisplayName("DELETE /user/favorites/all/subscriptions/{searchId}")
@RunWith(GuiceTestRunner.class)
@GuiceModules(PublicApiSearchesModule.class)
public class DeleteSubscriptionTest {

    @Rule
    @Inject
    public RuleChain defaultRules;

    @Inject
    private ApiClient api;

    @Inject
    private Account account;

    @Inject
    private PublicApiAdaptor adaptor;

    @Test
    public void shouldSee403WhenNoAuth() {
        String sessionId = adaptor.login(account).getSession().getId();
        String id = adaptor.addSearch(CARS, sessionId).getId();
        api.userFavorites().deleteSavedSearch().searchIdPath(id)
                .execute(validatedWith(shouldBeCode(SC_FORBIDDEN)));
    }

    @Test
    @Ignore("Создаем сессию")
    public void shouldSee401WithoutSessionId() {
        String sessionId = adaptor.login(account).getSession().getId();
        String id = adaptor.addSearch(CARS, sessionId).getId();
        api.userFavorites().deleteSavedSearch().searchIdPath(id)
                .reqSpec(defaultSpec()).execute(validatedWith(shouldBe401NeedAuthentication()));
    }
}

File: classfields/public-api-tests/src/test/java/ru/auto/tests/publicapi/lenta/DeleteSubscriptionTest.scala
package ru.auto.tests.publicapi.lenta

import com.carlosbecker.guice.{GuiceModules, GuiceTestRunner}
import com.google.inject.Inject
import io.qameta.allure.junit4.DisplayName
import org.apache.http.HttpStatus.{SC_BAD_REQUEST, SC_FORBIDDEN, SC_NOT_FOUND}
import org.junit.rules.RuleChain
import org.junit.runner.RunWith
import org.junit.{Rule, Test}
import ru.auto.tests.commons.restassured.ResponseSpecBuilders.shouldBe200OkJSON
import ru.auto.tests.publicapi.ApiClient
import ru.auto.tests.publicapi.ResponseSpecBuilders.{shouldBeCode, validatedWith}
import ru.auto.tests.publicapi.adaptor.PublicApiAdaptor
import ru.auto.tests.publicapi.model.AutoLentaSubscription
import ru.auto.tests.publicapi.module.PublicApiModule
import ru.auto.tests.publicapi.ra.RequestSpecBuilders.defaultSpec

import scala.annotation.meta.getter
import scala.jdk.CollectionConverters._

@DisplayName("DELETE lenta/delete-subscription")
@GuiceModules(Array(classOf[PublicApiModule]))
@RunWith(classOf[GuiceTestRunner])
class DeleteSubscriptionTest {

  @(Rule@getter)
  @Inject
  val defaultRules: RuleChain = null

  @Inject
  private val api: ApiClient = null

  @Inject
  private val adaptor: PublicApiAdaptor = null

  @Test
  def deleteNonExistingSubscription(): Unit = {
    val userId = "user:18"

    val tags = List("NISSAN", "NISSAN#MICRA", "NISSAN#MICRA#7758163")
    setSubscriptionTags(userId, tags)
    adaptor.deleteLentaSubscriptions(userId)
    adaptor.getLentaSubscriptions(userId, SC_NOT_FOUND)

    api.lenta().deleteSubscriptionLenta()
      .reqSpec(defaultSpec)
      .userIdQuery(userId)
      .execute(validatedWith(shouldBe200OkJSON))

    adaptor.getLentaSubscriptions(userId, SC_NOT_FOUND)
  }

  @Test
  def shouldSee403WhenNoAuth(): Unit = {
    api.lenta().deleteSubscriptionLenta()
      .execute(validatedWith(shouldBeCode(SC_FORBIDDEN)))
  }

  @Test
  def shouldSee400OnInvalidUserId(): Unit = {
    val userId = "invalid_user:123"
    api.lenta().deleteSubscriptionLenta()
      .reqSpec(defaultSpec)
      .userIdQuery(userId)
      .execute(validatedWith(shouldBeCode(SC_BAD_REQUEST)))
  }

  private def setSubscriptionTags(userId: String, tagsToAdd: List[String]) = {
    val addedTagsSubscription =
      new AutoLentaSubscription().includeTags(tagsToAdd.asJava)
    adaptor.addLentaSubscription(userId, addedTagsSubscription)
  }
}
